# dev-libs/libgpg-error: cross compiling fails with "src/syscfg/lock-obj-pub.<tuple>.h': No such file or directory".
# https://bugs.gentoo.org/584052

diff --git a/src/syscfg/lock-obj-pub.aarch64-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.aarch64-unknown-linux-musl.h
new file mode 100644
index 0000000..adf10fc
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.aarch64-unknown-linux-musl.h
@@ -0,0 +1,26 @@
+## lock-obj-pub.aarch64-unknown-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[48];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.aarch64_be-unknown-linux-gnu.h b/src/syscfg/lock-obj-pub.aarch64_be-unknown-linux-gnu.h
new file mode 100644
index 0000000..adf10fc
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.aarch64_be-unknown-linux-gnu.h
@@ -0,0 +1,26 @@
+## lock-obj-pub.aarch64-unknown-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[48];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.aarch64_be-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.aarch64_be-unknown-linux-musl.h
new file mode 100644
index 0000000..adf10fc
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.aarch64_be-unknown-linux-musl.h
@@ -0,0 +1,26 @@
+## lock-obj-pub.aarch64-unknown-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[48];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.arm-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.arm-unknown-linux-musl.h
new file mode 100644
index 0000000..7a92276
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.arm-unknown-linux-musl.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.arm-unknown-linux-gnueabi.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.armeb-unknown-linux-gnueabi.h b/src/syscfg/lock-obj-pub.armeb-unknown-linux-gnueabi.h
new file mode 100644
index 0000000..7a92276
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.armeb-unknown-linux-gnueabi.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.arm-unknown-linux-gnueabi.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.armeb-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.armeb-unknown-linux-musl.h
new file mode 100644
index 0000000..7a92276
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.armeb-unknown-linux-musl.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.arm-unknown-linux-gnueabi.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.i686-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.i686-unknown-linux-musl.h
new file mode 100644
index 0000000..b97273b
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.i686-unknown-linux-musl.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.i686-pc-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.mips-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.mips-unknown-linux-musl.h
new file mode 100644
index 0000000..b31206e
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.mips-unknown-linux-musl.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.mips-unknown-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff --git a/src/syscfg/lock-obj-pub.mipsel-unknown-linux-musl.h b/src/syscfg/lock-obj-pub.mipsel-unknown-linux-musl.h
new file mode 100644
index 0000000..3a24571
--- /dev/null
+++ b/src/syscfg/lock-obj-pub.mipsel-unknown-linux-musl.h
@@ -0,0 +1,23 @@
+## lock-obj-pub.mipsel-unknown-linux-gnu.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
